Gems/Atom/RPI/Code/Include/Atom/RPI.Public/MeshDrawPacket.h

```diff
@@ -89,6 +89,13 @@ namespace AZ
             bool DoUpdate(const Scene& parentScene);
             void ForValidShaderOptionName(const Name& shaderOptionName, const AZStd::function<bool(const ShaderCollection::Item&, ShaderOptionIndex)>& callback);
 
+            // To enable this shader constant in DrawSrg, a shader must:
+            // #define USE_DRAWSRG_MESHLOD_MESHINDEX 1
+            // By default it is NOT defined.
+            // When defined, the value of @m_modelLodMeshIndex (aka subMesh index) is written
+            // to the shader constant.
+            static constexpr AZStd::string_view DrawSrgModelLodMeshIndex = "m_modelLodMeshIndex";
```

Comment:
use `ShaderInputNameIndex` here, since its meant to deal with repeated validity checks

Reply:
I didn't want to increase the memory footprint of MeshDrawPacket, and followed what was being done with `UvStreamTangentBitmask::SrgName`. Especially for another opt-in shader constant that 99% of the time won't be used..

Changed it to `ShaderInputNameIndex` per your request, but I'm not sure it's the right thing, other than the codes looks cleaner.